File: 3095-shortest-subarray-with-or-at-least-k-i/3095-shortest-subarray-with-or-at-least-k-i.java
class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        int count = Integer.MAX_VALUE;
        for(int i = 0;i<nums.length;i++){
            int total = 0;
            for(int j = i;j<nums.length;j++){
                total = total | nums[j];
                if(total >= k){
                    count = Math.min(count, j - i + 1);
                }
            }
        }
        return count == Integer.MAX_VALUE ? -1 : count;
    }
}

File: 3095-shortest-subarray-with-or-at-least-k-i/3095-shortest-subarray-with-or-at-least-k-i.cpp
class Solution {
public:
    int minimumSubarrayLength(vector<int>& nums, int k) {
        int n = nums.size();
        int minLen = INT_MAX;
        for (int i = 0; i < n; i++) {
            int tempVal = 0;
            for (int j = i; j < n; ++j) {
                tempVal |= nums[j];
                if (tempVal >= k) {
                    minLen = min(minLen, j - i + 1);
                    break;
                }
            }
        }
        return minLen == INT_MAX ? -1 : minLen;
    }
};